rnote-compose/src/shapes/arrow.rs

```diff
@@ -0,0 +1,212 @@
+
+                Line {
+                    start: sub_start,
+                    end: sub_end,
+                }
+            })
+            .collect::<Vec<Line>>()
+    }
+}
+
+/// Functions which you can use to interact with the struct.
+impl Arrow {
+    /// The min-length for `rline` and `lline`.
+    const TIP_LINES_MIN_LENGTH: f64 = 32.0;
+
+    /// Creating a new arrow with the given start and tip vectors.
+    pub fn new(start: na::Vector2<f64>, tip: na::Vector2<f64>) -> Self {
+        Self { start, tip }
+    }
+
+    /// convert the arrow to kurbo-elements.
+    pub fn to_kurbo(&self, stroke_width: Option<f64>) -> ArrowKurbo {
+        let main = kurbo::Line::new(self.start.to_kurbo_point(), self.tip.to_kurbo_point());
+        let tip_triangle = {
+            let tip = self.tip.to_kurbo_point();
+            let lline = self.compute_lline(stroke_width).to_kurbo_point();
+            let rline = self.compute_rline(stroke_width).to_kurbo_point();
+
+            kurbo::BezPath::from_vec(vec![
+                PathEl::MoveTo(lline),
+                PathEl::LineTo(tip),
+                PathEl::LineTo(rline),
+            ])
+        };
+
+        ArrowKurbo {
+            stem: main,
+            tip_triangle,
+        }
+    }
+}
+
+/// This implementation holds the functions to get the vectors `rline` and
+/// `lline`.
+impl Arrow {
+    /// The angle for `rline` and `lline` to the stem of the arrow.
+    const ANGLE: Radian = (13.0 / 16.0) * std::f64::consts::PI;
+
+    /// The default direction vector (the stem) if you can't compute one.
+    const DEFAULT_DIRECTION_VECTOR: na::Vector2<f64> = na::Vector2::new(1.0, 0.0);
+
+    /// Computes and returns `lline`.
+    /// Optionally add the stroke width to adjust the length of the line.
+    pub fn compute_lline(&self, stroke_width: Option<f64>) -> na::Vector2<f64> {
+        let vec_a = self.get_direction_vector() * Self::get_line_length(stroke_width);
+        let rotation_matrix = self.get_rotation_matrix();
+
+        rotation_matrix * vec_a + self.tip
+    }
+
+    /// Computes and returns `rline`.
+    /// Optionally add the stroke width to adjust the length of the line.
+    pub fn compute_rline(&self, stroke_width: Option<f64>) -> na::Vector2<f64> {
+        let vec_b = self.get_direction_vector() * Self::get_line_length(stroke_width);
+        let rotation_matrix = self.get_rotation_matrix().transpose();
+
+        rotation_matrix * vec_b + self.tip
+    }
+
+    /// Returns the normalized direction vector from `start` to `tip`.
+    fn get_direction_vector(&self) -> na::Vector2<f64> {
+        let direction_vector = self.tip - self.start;
+
+        if direction_vector.norm() == 0.0 {
+            Self::DEFAULT_DIRECTION_VECTOR
+        } else {
+            direction_vector / direction_vector.norm()
+        }
+    }
+
+    /// Returns the rotation matrix for the tip lines how they should be
+    /// rotated.
+    fn get_rotation_matrix(&self) -> Rotation2<f64> {
```

Comment:
same here, although I would even argue that this function does too little to be one. I think we could inline `Rotation2::new()` where needed

Reply:
hm... in my opinion the name of the function does make it more readable. So, you insist to inline this function with `Rotation2::new()`?

Reply:
yeah just use Rotation2 directly